### PR TITLE
Align MongoDB Realm API's with other SDK's

### PR DESCRIPTION
**Pull request by clementetb**
Refactors MongoDB Realm API to match JS and Cocoa SDKs

### CODE REVIEW

File: realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SessionTests.kt [hunk 1]
```diff
@@ -502,5 +502,18 @@ class SessionTests {
...
+    fun getAllSessions(){
+        val realm = Realm.getInstance(configuration)
+        val session = realm.syncSession
+        val sessions = app.sync.allSessions
+
+        assertNotNull(sessions)
+        assertEquals(1, sessions.size)
+
+        realm.close()
+        session.stop()
```
Comment (cmelchior) [inline realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SessionTests.kt:516]: No need to manually stop the session. That should happen when closing the Realm

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java [hunk 2]
```diff
@@ -330,6 +330,20 @@ public User switchUser(User user) {
...
+     * Removes a users credentials from this device. If the user was currently logged in, they
+     * will be logged out as part of the process. This is only a local change and does not
+     * affect the user state on the server.
+     *
+     * @param user to remove
+     * @return user that was removed.
+     * @throws AppException if called from the UI thread or if the user was logged in, but
+     *                      could not be logged out.
+     */
+    public User removeUser(User user) throws AppException {
```
Comment (cmelchior) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java:343]: We also need to move the `removeUserAsync()` method.

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/Profile.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+package io.realm.mongodb;
+
+import javax.annotation.Nullable;
+
+import io.realm.internal.objectstore.OsSyncUser;
+
+public class Profile {
+    private final OsSyncUser osUser;
+
+    public Profile(OsSyncUser osUser) {
```
Comment (cmelchior) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/Profile.java:26]: This constructor shouldn't be public if possible. Also, we should probably send in the `User` class as a parameter as we can then expose it through a `getUser()` (for completeness, so we can always find the user/app from any class in the class hierachy)

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/Profile.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+ * limitations under the License.
+ */
+
+package io.realm.mongodb;
+
+import javax.annotation.Nullable;
+
+import io.realm.internal.objectstore.OsSyncUser;
+
+public class Profile {
```
Comment (cmelchior) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/Profile.java:23]: We should probably implement equals/hashcode/toString

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/mongo/MongoClient.java [hunk 5]
```diff
@@ -48,4 +48,13 @@ public MongoDatabase getDatabase(final String databaseName) {
         return new MongoDatabase(osMongoClient.getDatabase(databaseName, codecRegistry),
                 databaseName);
     }
+
+    /**
+     * Returns the service name for this client.
+     *
+     * @return the service name.
+     */
+    public String getServiceName() {
```
Comment (cmelchior) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/mongo/MongoClient.java:57]: We seem to be lacking tests for this?

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/mongo/MongoCollection.java [hunk 6]
```diff
@@ -85,6 +85,15 @@ public MongoNamespace getNamespace() {
         return nameSpace;
     }
 
+    /**
+     * Gets the name of this collection
+     *
+     * @return the name
+     */
+    public String getName() {
```
Comment (cmelchior) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/mongo/MongoCollection.java:93]: We seem to be lacking tests for this?

**Review by cmelchior — CHANGES_REQUESTED** [review]
Nice 👏 . Left a few minor comments, but we also need to document all of these changes in the CHANGELOG under breaking changes.

File: realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SessionTests.kt (hunk 1 above)
Comment (clementetb) [reply realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SessionTests.kt:516]: Ok!

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java (hunk 2 above)
Comment (clementetb) [reply realm/realm-library/src/objectServer/java/io/realm/mongodb/App.java:343]: Didn't see the method 🤦🏻‍♂️

**Review by edualonso — APPROVED** [review]
Looks good.

**Review by rorbech — CHANGES_REQUESTED** [review]
Looks good, but still lacks listing the breaking changes in the CHANGELOG

File: realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/UserProfileInfoTests.kt [hunk 7]
```diff
@@ -29,7 +29,7 @@ import org.junit.runner.RunWith
 import kotlin.test.assertNull
 
 @RunWith(AndroidJUnit4::class)
-class UserProfileTests {
+class UserProfileInfoTests {
```
Comment (rorbech) [inline realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/UserProfileInfoTests.kt:32]: I have not picked up anything about changing Profile to ProfileInfo. Is this aligned with the other SDKs?

**Review by rorbech — APPROVED** [review]
Only minor comment, otherwise looks good.

File: CHANGELOG.md [hunk 8]
```diff
@@ -5,12 +5,30 @@ We no longer support Realm Cloud (legacy), but instead the new MongoDB Realm Clo
...
+* [RealmApp] Renamed `ApiKeyAuth.fetchAllApiKeys()` to `ApiKeyAuth.fetchAll()` and `ApiKeyAuth.fetchAllApiKeysAsync()` to `ApiKeyAuth.fetchAllAsync()`.
+* [RealmApp] Renamed `ApiKeyAuth.deleteApiKey()` to `ApiKeyAuth.delete()` and `ApiKeyAuth.deleteApiKeyAsync()` to `ApiKeyAuth.deleteAsync()`.
+* [RealmApp] Renamed `ApiKeyAuth.enableApiKey()` to `ApiKeyAuth.enable()` and `ApiKeyAuth.enableApiKeyAsync()` to `ApiKeyAuth.enableAsync()`.
+* [RealmApp] Renamed `ApiKeyAuth.disableApiKey()` to `ApiKeyAuth.disable()` and `ApiKeyAuth.disableApiKeyAsync()` to `ApiKeyAuth.disableAsync()`.
+* [RealmApp] Renamed `User.getApiKeysAuth()` to `User.getApiKeys()`.
+* [RealmApp] Renamed `UserApiKey` class to `ApiKey`.
+* [RealmApp] Removed support for `Credentials.serverApiKey()`.
+* [RealmApp] Renamed `App.getEmailPasswordAuth()` to `App.getEmailPassword()`.
+* [RealmApp] User profile methods `getName()`, `getEmail()`, `getPictureUrl()`, `getFirstName()`, `getLastName()`, `getGender()`, `getBirthday()`, `getMinAge()` and `getMaxAge()` are now available under a new class ProfileInfo. It can be accessed using `User.getProfile()`.
```
Comment (cmelchior) [inline CHANGELOG.md:19]: My bad, but I remembered wrong. JS is calling this class `UserProfile` so we should rename ours to the same. Sorry.

File: realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/UserProfileInfoTests.kt (hunk 7 above)
Comment (cmelchior) [reply realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/UserProfileInfoTests.kt:32]: Sorry, I remembered wrong. JS is calling this `UserProfile` so we should rename to that as well: https://mongodb.slack.com/archives/CKW2HG3M0/p1599466293136100

**Review by cmelchior — APPROVED** [review]

**Review by rorbech — APPROVED** [review]

**Pull request merged**